### PR TITLE
[Shapes] Added full support for intaking percentage values in the internal shape generation

### PR DESCRIPTION
**Pull request by yarneo**
This is a follow up PR to #5090, #5091  and contains all of their changes as well until that PR is merged.

This PR finalizes the percentage support and adds the algorithm to provide percentage values to be set on the shape's path.

### CODE REVIEW

**Comment by yarneo** [comment]
This PR is no longer blocked

File: components/private/Shapes/src/MDCCornerTreatment.h [hunk 1]
```diff
@@ -59,4 +59,16 @@ typedef NS_ENUM(NSInteger, MDCCornerTreatmentValueType) {
  */
 - (nonnull MDCPathGenerator *)pathGeneratorForCornerWithAngle:(CGFloat)angle;
 
+/**
+ Generates an MDCPathGenerator object for a corner with the provided angle.
```
Comment (jverkoey) [inline components/private/Shapes/src/MDCCornerTreatment.h]: Suggestion: `Creates` to remove redundant "generates a generator" language.

**Review by jverkoey — COMMENTED** [review]
There are a fair number of tricky bits of logic here - how difficult will it be to write tests?

**Comment by yarneo** [comment]
Wasn't too easy to write this test, but managed to get a good test going!

File: components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m [hunk 2]
```diff
@@ -102,6 +103,40 @@ - (void)testPercentageValueEqualityForCorners {
...
+  // The outcome of an 100x100 square with 50% cut corners is a diamond with points at
+  // (0, 50), (50, 0), (100, 50), (50, 100)
+  XCTAssertEqual([pathPoints count], 8);
+  for (NSValue *value in pathPoints) {
+    CGPoint point = value.CGPointValue;
+    XCTAssert(MDCCGFloatEqual(MDCRound(point.x), 0.f) || MDCCGFloatEqual(MDCRound(point.x), 50.f) ||
+              MDCCGFloatEqual(MDCRound(point.x), 100.f));
+    XCTAssert(MDCCGFloatEqual(MDCRound(point.y), 0.f) || MDCCGFloatEqual(MDCRound(point.y), 50.f) ||
+              MDCCGFloatEqual(MDCRound(point.y), 100.f));
+  }
```
Comment (jverkoey) [inline components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m]: Suggestion: compare against an array of points.

**Review by jverkoey — APPROVED** [review]

File: components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m (hunk 2 above)
Comment (yarneo) [reply components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m]: done

**Pull request merged**